File: tests/typesafety/python/transfer_arguments.py
# expect: passes
# title: Two account numbers share one type, so a refund can flow the wrong way
def refund(source_account: str, target_account: str, amount_cents: int) -> str:
    return f"move {amount_cents} from {source_account} to {target_account}"


def refund_supplier(shop_account: str, supplier_account: str) -> str:
    # The arguments are in the wrong order. The checker accepts the call,
    # because both parameters are the same type, and the money flows into
    # the shop instead of back to the supplier.
    return refund(supplier_account, shop_account, 4_500)

File: tests/typesafety/typescript/transfer_arguments.ts
// expect: passes

function transfer(fromAccount: string, toAccount: string, amountCents: number): string {
  return `move ${amountCents} from ${fromAccount} to ${toAccount}`;
}

export function payRent(tenantAccount: string, landlordAccount: string): string {
  // The arguments are in the wrong order. The checker accepts this call,
  // because both parameters have the same type. The money goes the wrong way.
  return transfer(landlordAccount, tenantAccount, 95_000);
}
